src/cheri-pte-ext.adoc

```diff
@@ -10,9 +10,13 @@ capabilities in memory at the page granularity. For this reason, the
 {cheri_pte_ext_name} extension adds new bits to RISC-V's Page Table Entry (PTE)
 format.
 
-Implementing any virtual memory translation scheme (_Sv39_, _Sv48_ or _Sv57_) and {cheri_base_ext_name} requires {cheri_pte_ext_name} to be implemented.
+NOTE: There is no explicit mechanism for enabling or disabling {cheri_pte_ext_name}. A VM-enabled legacy (non-CHERI) OS running in {cheri_int_mode_name} will not load or store capabilities, and so the default state of CW=0 causing loaded capabilities to have their tags cleared, and stored capabilities with their tags set to cause a page fault, won't occur.
 
-NOTE: There is no explicit mechanism for enabling or disabling {cheri_pte_ext_name}. A VM-enabled legacy (non-CHERI) OS running in {cheri_int_mode_name} will not load or store capabilities, and so the default state of CW=0 causing loaded capabilities to have their tags cleared, and stored capabilities with their tags set to cause a page fault, won't occur. A CHERI-aware OS running a VM-enabled OS is required to support {cheri_pte_ext_name}, and the minimum level of support is to set CW to 1 in all PTEs and leave <<sstatusreg_pte,sstatus>>.CRG and CRG in all PTEs set to 0, which will allow capabilities with their tags set to be loaded and stored successfully.
+A CHERI-aware OS running a VM-enabled OS on an RV64 core is required to support {cheri_pte_ext_name}, and the minimum level of support is to set CW to 1 in all PTEs intended for storing capabilities (i.e. anonymous mappings) and leave <<sstatusreg_pte,sstatus>>.CRG and CRG in all PTEs set to 0, which will allow capabilities with their tags set to be loaded and stored successfully.
+
+Therefore Implementing any RV64 virtual memory translation scheme (_Sv39_, _Sv48_ or _Sv57_) and {cheri_base_ext_name} requires {cheri_pte_ext_name} to be implemented.
```

Comment:
But based on your comments on the other PR the problem is there is no mechanism to detect whether the extension is present. So I guess the fix for that is to require it - although it has a different stability level from the other extension so not sure if we can do that right now.

Reply:
well -that's a good point - all the more reason to get on and test it!
but it's simple enough to set CW=1 in all PTEs, and add full support later, so I guess the risk is minimised.

Reply:
I guess you can always probe for a trap. Not the nicest way of detecting presence of the extension but it would work.

This means that we can dynamically detect presence of the extension, so it should be legal to not implement it (although I'd prefer if everyone did).

Reply:
If there is a way to detect it then best add that text. We could make it strongly recommended then.

Reply:
You would need to do so extremely early in boot. It'd be a big ugly wart in the code to do so. Just because you technically can doesn't mean you should.

Reply:
text updated, please have another look